src/components/Options/index.tsx

```diff
@@ -61,25 +67,68 @@ export default class Options extends Component<Props, State> {
   }
 
   @bind
-  onTypeChange(event: Event) {
+  onEncoderTypeChange(event: Event) {
     const el = event.currentTarget as HTMLSelectElement;
 
     // The select element only has values matching encoder types,
     // so 'as' is safe here.
     const type = el.value as EncoderType;
-    this.props.onTypeChange(type);
+    this.props.onEncoderTypeChange(type);
+  }
+
+  @bind
+  onPreprocessorEnabledChange(event: Event) {
+    const el = event.currentTarget as HTMLInputElement;
+
+    const preprocessorState = this.props.preprocessorState;
+    const preprocessor = el.name.split('.')[0] as keyof typeof preprocessorState;
+    preprocessorState[preprocessor].enabled = el.checked;
+    this.props.onPreprocessorOptionsChange(preprocessorState);
+  }
+
+  @bind
+  onQuantizerOptionsChange(opts: QuantizeOptions) {
```

Comment:
There’s probably a nice way of writing a generic `onPreprocessorOptionsChange` handler, but I’d liek to address that in a separate PR.

Reply:
#118